File: RestaurantFrontEnd/ClientApp/src/app/models/ordered-item.ts
import { Entity } from "./entity";
import { Order } from "./order";
import { Item } from "./item";
export class OrderedItem extends Entity{
  Order: Order;
  Item: Item;
  Quantity: number;
  Price: number;
  Subtotal: number;
  Remarks: number;

  constructor(params: OrderedItem = {} as OrderedItem)
  {
    super(params);
    this.Order = params.Order;
    this.Item = params.Item;
    if (params.Price == null && params.Item != null) { this.Price = params.Item.Price; } else { this.Price = params.Price; }
    this.Quantity = params.Quantity;
    this.Subtotal = this.Price * this.Quantity;
    this.Remarks = params.Remarks;

  }
}

File: RestaurantDataModel/Model/Entity/OrderedItem.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;


namespace Restaurant.DataModel.Entity {

    public class OrderedItem : Entity, IEntity {
       
        [JsonSerializeAttribute]
        public virtual System.Guid Id { get; set; }
        [JsonSerializeAttribute]
        public virtual FoodOrder Order { get; set; }
        [JsonSerializeAttribute]
        public virtual Item Item { get; set; }
        [JsonSerializeAttribute]
        [Required]
        public virtual byte Quantity { get; set; }
    
            [JsonSerializeAttribute]
        [Required]
        public virtual decimal Price { get; set; }
            [JsonSerializeAttribute]
        [Required]
        public virtual decimal Subtotal { get; set; }
            [JsonSerializeAttribute]
        [StringLength(2147483647)]
        public virtual string Remarks { get; set; }
            public virtual void Compute()
            {
                Subtotal = Decimal.Multiply(Price, Quantity);
            }
        }
}
